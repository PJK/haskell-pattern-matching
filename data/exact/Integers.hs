module Integers where

-- odd :: Int -> Boolean
-- odd 1 = True
-- odd 2 = False
-- Infinite set of uncovered constructors
-- This parses to LiteralPattern