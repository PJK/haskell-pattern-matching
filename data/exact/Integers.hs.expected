{
    "tag": "AnalysisSuccess",
    "contents": [
        [
            "odd",
            {
                "tag": "NonExhaustive",
                "contents": [
                    [
                        {
                            "tag": "VariablePattern",
                            "contents": "fresh0"
                        }
                    ],
                    [
                        {
                            "tag": "VariablePattern",
                            "contents": "fresh0"
                        }
                    ],
                    [
                        {
                            "tag": "VariablePattern",
                            "contents": "fresh0"
                        }
                    ],
                    [
                        {
                            "tag": "VariablePattern",
                            "contents": "fresh0"
                        }
                    ]
                ]
            }
        ]
    ]
}